### PR TITLE
Web deployment refactored

### PR DESCRIPTION
**Pull request by Imipenem**

### CODE REVIEW

File: cookietemple/create/templates/web/website_python/flask/{{ cookiecutter.project_slug_no_hyphen }}/deployment_scripts/setup.sh [hunk 1]
```diff
@@ -1,29 +1,42 @@
...
-cp /home/{{cookiecutter.vmusername}}/{{cookiecutter.project_slug}}/deployment_scripts/{{cookiecutter.project_slug_no_hyphen}}.service \
+sudo pip3 install virtualenv
+virtualenv dpenv
+source dpenv/bin/activate
+
+sudo apt-get update
+pip3 install gunicorn
+make install
+
+sudo cp /home/{{cookiecutter.vmusername}}/{{cookiecutter.project_slug}}/deployment_scripts/{{cookiecutter.project_slug_no_hyphen}}.service \
```
Comment (Zethson) [inline cookietemple/create/templates/web/website_python/flask/{{ cookiecutter.project_slug_no_hyphen }}/deployment_scripts/setup.sh:18]: Does this require sudo?

File: cookietemple/create/templates/web/website_python/flask/{{ cookiecutter.project_slug_no_hyphen }}/deployment_scripts/setup.sh [hunk 2]
```diff
@@ -1,29 +1,42 @@
...
 /etc/systemd/system/{{cookiecutter.project_slug_no_hyphen}}.service
 
-systemctl start {{cookiecutter.project_slug}}
+sudo systemctl start {{cookiecutter.project_slug}}
 
-systemctl enable {{cookiecutter.project_slug}}
+sudo systemctl enable {{cookiecutter.project_slug}}
 
-cp /home/{{cookiecutter.vmusername}}/{{cookiecutter.project_slug}}/deployment_scripts/{{cookiecutter.project_slug_no_hyphen}} \
+sudo cp /home/{{cookiecutter.vmusername}}/{{cookiecutter.project_slug}}/deployment_scripts/{{cookiecutter.project_slug_no_hyphen}} \
```
Comment (Zethson) [inline cookietemple/create/templates/web/website_python/flask/{{ cookiecutter.project_slug_no_hyphen }}/deployment_scripts/setup.sh:25]: does this require sudo?

File: cookietemple/create/templates/web/website_python/flask/{{ cookiecutter.project_slug_no_hyphen }}/deployment_scripts/setup.sh [hunk 3]
```diff
@@ -1,29 +1,42 @@
...
 /etc/nginx/sites-available/{{cookiecutter.project_slug_no_hyphen}}
 
-ln -s /etc/nginx/sites-available/{{cookiecutter.project_slug_no_hyphen}} /etc/nginx/sites-enabled
+sudo ln -s /etc/nginx/sites-available/{{cookiecutter.project_slug_no_hyphen}} /etc/nginx/sites-enabled
```
Comment (Zethson) [inline cookietemple/create/templates/web/website_python/flask/{{ cookiecutter.project_slug_no_hyphen }}/deployment_scripts/setup.sh:28]: sudo?

File: cookietemple/create/templates/web/website_python/flask/{{ cookiecutter.project_slug_no_hyphen }}/deployment_scripts/setup.sh [hunk 4]
```diff
@@ -1,29 +1,42 @@
...
 
-nginx -t
+sudo nginx -t
 
-systemctl restart nginx
+sudo systemctl restart nginx
 
-ufw delete allow 5000
+sudo ufw delete allow 5000
```
Comment (Zethson) [inline cookietemple/create/templates/web/website_python/flask/{{ cookiecutter.project_slug_no_hyphen }}/deployment_scripts/setup.sh:34]: required? below as well?

File: cookietemple/create/templates/web/website_python/flask/{{ cookiecutter.project_slug_no_hyphen }}/deployment_scripts/setup.sh [hunk 5]
```diff
@@ -1,29 +1,42 @@
 #!/bin/bash
 # Reference:
 # https://www.digitalocean.com/community/tutorials/how-to-serve-flask-applications-with-gunicorn-and-nginx-on-ubuntu-18-04
+sudo apt-get install software-properties-common
+sudo apt-add-repository universe
+sudo apt-get update
+sudo apt-get install python3-pip
```
Comment (Zethson) [inline cookietemple/create/templates/web/website_python/flask/{{ cookiecutter.project_slug_no_hyphen }}/deployment_scripts/setup.sh:7]: combine with below

Comment (Imipenem) [reply cookietemple/create/templates/web/website_python/flask/{{ cookiecutter.project_slug_no_hyphen }}/deployment_scripts/setup.sh:7]: With below?

File: cookietemple/create/templates/web/website_python/flask/{{ cookiecutter.project_slug_no_hyphen }}/deployment_scripts/setup.sh (hunk 1 above)
Comment (Imipenem) [reply cookietemple/create/templates/web/website_python/flask/{{ cookiecutter.project_slug_no_hyphen }}/deployment_scripts/setup.sh:18]: May not, but didn't have the time to test it, will look at it when deploying my website

File: cookietemple/create/templates/web/website_python/flask/{{ cookiecutter.project_slug_no_hyphen }}/deployment_scripts/setup.sh (hunk 5 above)
Comment (Zethson) [reply cookietemple/create/templates/web/website_python/flask/{{ cookiecutter.project_slug_no_hyphen }}/deployment_scripts/setup.sh:7]: the line below

**Pull request merged**